c-ext/compressionwriter.c

```diff
@@ -290,6 +290,8 @@ static PyMethodDef ZstdCompressionWriter_methods[] = {
      PyDoc_STR("Enter a compression context.")},
     {"__exit__", (PyCFunction)ZstdCompressionWriter_exit, METH_VARARGS,
      PyDoc_STR("Exit a compression context.")},
+    {"__iter__", (PyCFunction)ZstdCompressionWriter_unsupported, METH_NOARGS, NULL},
+    {"__next__", (PyCFunction)ZstdCompressionWriter_unsupported, METH_NOARGS, NULL},
```

Comment:
Unfortunately, this isn't the correct way to define `__iter__` and `__next__` on C types. Since `__iter__` and `__next__` are part of what the C API refers to as a `protocol`, these should be defined via the `tp_*` slots on the `PyTypeObject` type definition.

I'll refactor this locally.

And, the test failures are due to adding tests for the `ZstdDecompressionWriter` but not implementing these methods there.